common/beerocks/hostapd/configuration.cpp

```diff
@@ -151,6 +151,37 @@ bool Configuration::set_create_vap_value(const std::string &vap, const std::stri
     return set_create_vap_value(vap, key, std::to_string(value));
 }
 
+std::string Configuration::get_vap_value(const std::string &vap, const std::string &key)
+{
+    // search for the requested vap
+    auto existing_vap = m_hostapd_config_vaps.find(vap);
+
+    if (existing_vap == m_hostapd_config_vaps.end()) {
+        m_last_message = std::string(__FUNCTION__) + " couldn't find requested vap: " + vap +
+                         " (requested key: " + key + ")";
+        m_ok = false;
```

Comment:
Hm, I don't really like this... A missing VAP doesn't mean that the configuration suddenly becomes invalid. IMHO it would be more logical to return bool and have an output argument for the value. The caller will anyway need a condition to check if the key was present, so it's not possible to chain this function into another operation.

(As usual, it would be a lot easier if we could throw an exception...)